hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -768,28 +815,28 @@ def __init__(self, max_length, prefix, random, observer=None):
         self.testcounter = global_test_counter
         global_test_counter += 1
         self.start_time = time.perf_counter()
-        self.events = set()
-        self.forced_indices = set()
+        self.events: "Union[Set[str], FrozenSet[str]]" = set()
+        self.forced_indices: "Set[int]" = set()
         self.interesting_origin = None
-        self.draw_times = []
+        self.draw_times: "List[float]" = []
         self.max_depth = 0
         self.has_discards = False
 
-        self.__result = None
+        self.__result: "Optional[ConjectureResult]" = None
 
         # Observations used for targeted search.  They'll be aggregated in
         # ConjectureRunner.generate_new_examples and fed to TargetSelector.
         self.target_observations = {}
```

Comment:
Any suggestions for the type here?

Reply:
`Dict[Optional[str], Union[int, float]]` - it's the backing data from `hypothesis.target()`.